common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/OneForOneBlockFetcher.java

```diff
@@ -58,29 +59,41 @@
   private final BlockFetchingListener listener;
   private final ChunkReceivedCallback chunkCallback;
   private TransportConf transportConf = null;
-  private File[] shuffleFiles = null;
+  private Boolean toDisk;
+  private Supplier<File> tmpFileCreater;
+  private Supplier<Boolean> shuffleBlockFetcherIteratorIsZombie;
 
   private StreamHandle streamHandle = null;
 
   public OneForOneBlockFetcher(
+    TransportClient client,
+    String appId,
+    String execId,
+    String[] blockIds,
+    BlockFetchingListener listener,
+    TransportConf transportConf) {
+    this(client, appId, execId, blockIds, listener, transportConf, false, null, null);
+  }
+
+  public OneForOneBlockFetcher(
       TransportClient client,
       String appId,
       String execId,
       String[] blockIds,
       BlockFetchingListener listener,
       TransportConf transportConf,
-      File[] shuffleFiles) {
+      Boolean toDisk,
+      Supplier<File> tmpFileCreater,
+      Supplier<Boolean> shuffleBlockFetcherIteratorIsZombie) {
```

Comment:
Since the network module doesn't depend on the core module, we should hide the `ShuffleBlockFetcherIterator` from network module. How about we name it `canCallerSideDeleteFile`?